tests/test_georaster.py

```diff
@@ -195,6 +196,162 @@ def test_loading(self) -> None:
         assert geoutils.misc.array_equal(r.bands, (2, 3))
         assert r.data.shape == (r.nbands, r.height, r.width)
 
+    @pytest.mark.parametrize("nodata_init", [None, "type_default"])  # type: ignore
+    @pytest.mark.parametrize(
+        "dtype", ["uint8", "int8", "uint16", "int16", "uint32", "int32", "float32", "float64", "float128"]
+    )  # type: ignore
+    def test_data_setter(self, dtype: str, nodata_init: str | None) -> None:
+        """
+        Test that the behaviour of data setter, which is triggered directly using from_array, is as expected.
+        In details, we check that the data setter:
+
+        1. Writes the data in a masked array, whether the input is a classic array or a masked_array,
+        2. Reshapes the data in a 3D array if it is 2D,
+        3. Sets a new nodata value only if the provided array is not a masked_array and contains non-finite values,
+        4. Masks non-finite values that are unmasked, whether the input is a classic array or a masked_array,
+        """
+
+        nodata_init = None
+        dtype = "float32"
+
+        # Initiate a random array for testing
+        width = height = 5
+        transform = rio.transform.from_bounds(0, 0, 1, 1, width, height)
+
+        # Create random values between the lower and upper limit of the data type, max absolute 99999 for floats
+        if "int" in dtype:
+            val_min = np.iinfo(int_type=dtype).min
+            val_max = np.iinfo(int_type=dtype).max
+            randint_dtype = dtype
+        else:
+            val_min = -99999
+            val_max = 99999
+            randint_dtype = "int32"
+
+        # Fix the random seed
+        np.random.seed(42)
+        arr = np.random.randint(low=val_min, high=val_max, size=(1, width, height), dtype=randint_dtype).astype(dtype)
+        mask = np.random.randint(0, 2, size=(1, width, height), dtype=bool)
```

Comment:
You can also mask e.g. the min values, so they are random and you know they exist.

Reply:
Seeds are recommended for unitests even when they should pass for any random value, e.g. to reproduce locally if there is ever an error, so it works nicely like this